bindings/postgres/postgres.go

```diff
@@ -186,13 +186,21 @@ func (p *Postgres) query(ctx context.Context, sql string, args ...any) (result [
 		return nil, fmt.Errorf("error executing query: %w", err)
 	}
 
+    cols := rows.FieldDescriptions()
+
 	rs := make([]any, 0)
 	for rows.Next() {
 		val, rowErr := rows.Values()
 		if rowErr != nil {
 			return nil, fmt.Errorf("error reading result '%v': %w", rows.Err(), rowErr)
 		}
-		rs = append(rs, val) //nolint:asasalint
+		
+		r := map[string]interface{}{}
+		for i, col := range cols {
+			r[string(col.Name)] = val[i]
```

Comment:
Sadly, this is a breaking change. Currently, users expect to receive a slice, but with this, they will receive a map. This will break all existing users.

Perhaps this should be a configurable option?

Reply:
Infact, the query function return a json, not a map or a slice: "result, err = json.Marshal(rs)", and the json should include column name，like the query function in mysql.go: "result, err := m.jsonify(rows)"

Reply:
Yes, my point is: that is a breaking change. If users expect the JSON to contain an array, and you're now returning a dictionary, that will break applications.

Reply:
I agree with your opinion, and we expect the JSON contain a dictionary，how to make a configurable option? Can you help me?

Reply:
You need to do this via request metadata which a user would need to specify to enable your new functionality, or introduce a new operation code. You can look at the code of other bindings to see how this can be done.